src/lib/swapper/swappers/ThorchainSwapper/utils/getSignTxFromQuote.ts

```diff
@@ -63,7 +63,7 @@ export const getSignTxFromQuote = async ({
       const { router, data } = quote as ThorEvmTradeQuote
       const sendTxInput = await createSignTxInput({
         accountNumber,
-        from,
+        from: from!,
```

Comment:
We might want to be better at discrimination here, we use `xpub` for UTXOs, `from` for all others

Reply:
agreed, though end state could be enforcing these with zod in the api yeah?

Reply:
End state would be separated types per chain i.e instead of `& ({ from: string; xpub?: never } | { from?: never; xpub: string })):` type, we would have one input for UTXOs, one for others - with or without a generic, as we've decided these were too much bloat.
If we don't use generics, then this won't automagically be enforced to be one or the other depending on the chain, but could be mitigated by typing the consumers' inputs with the correct variant